crates/floresta-compact-filters/src/network_filters.rs

```diff
@@ -22,16 +22,28 @@ impl<Storage: IterableFilterStore> NetworkFilters<Storage> {
     pub fn match_any(
         &self,
         query: Vec<&[u8]>,
-        start_height: Option<usize>,
+        start_height: Option<u32>,
+        stop_height: Option<u32>,
         chain: impl BlockchainInterface,
     ) -> Result<Vec<BlockHash>, IterableFilterStoreError> {
         let mut blocks = Vec::new();
         let iter = query.into_iter();
+
+        let start_height: Option<usize> = match start_height {
+            Some(n) => Some(n as usize),
+            None => Some(0_usize),
+        };
+
         for (height, filter) in self.filters.iter(start_height)? {
             let hash = chain.get_block_hash(height).unwrap();
             if filter.match_any(&hash, &mut iter.clone()).unwrap() {
                 blocks.push(hash);
             }
+            if let Some(stop_at) = stop_height {
+                if height >= stop_at {
+                    break;
+                };
+            };
```

Comment:
You can create one `stop_height` above this to be either `stop_height` or `self.height`

Reply:
This avoids the `if let Some`